galaxy/tools/verify/test_data.py

```diff
@@ -78,7 +78,7 @@ def __init__(self, repository, environ):
         self.updated = False
         repo_cache = environ.get("GALAXY_TEST_DATA_REPO_CACHE", "test-data-cache")
         m = hashlib.md5()
-        m.update(repository)
+        m.update(repository.encode('utf-8'))
```

Comment:
What about `m.update(smart_str(repository))` using `smart_str` from `galaxy.util`? It's basically the inverse of `unicodify`.